src/processors/knex-processor.ts

```diff
@@ -132,7 +132,9 @@ export default class KnexProcessor<ResourceT extends Resource> extends Operation
     const { type, schema } = this.resourceClass;
     const { attributes, relationships, primaryKeyName } = schema;
     const relationshipsKeys = Object.entries(relationships)
-      .filter(([, value]) => value.belongsTo)
+      .filter(([, value]) => {
+        return value.belongsTo && value.alwaysIncludeLinkageData;
```

Comment:
@joelalejandro I hope this is the right place to exclude linkages.
I have to figure how to handle few cases.
1. If `alwaysIncludeLinkageData` is false we should skip the relationship query 
2. if `alwaysIncludeLinkageData` is false and url contains `include=comments` then it should include linkage
3. if `alwaysIncludeLinkageData` is false and url contains `fields[articles]=comments`
4. if `alwaysIncludeLinkageData` is true then always include linkages

It should also take care of deep nested relationships, but that over my head right now.
Is there any other case? I need help with this one...

Reply:
Also, it seems that 2. case is the most difficult because I don't have any info about include.

Reply:
I think you might want to make such decisions in the `JsonApiSerializer` class. In particular, `serializeResource` and `serializeIncludedResources`.

Reply:
I'm pretty confident that it should be done here. 
It's the matter fo optimisation. Why should we build a complex DB query if we are not showing those resources in the response. It seems like a waste of processing time. 
Also, toMany resources could be filtered, paginated and sorted which also could be skipped if we relationships has links enabled.

Reply:
Alright, I guess you could add a third parameter that allows you to bring extra context to the scope of `getColumns`. In the `get()` method of the processor, you have `op`, which is the operation being resolved in the current request. There, you can check `params.include`. The [get()](https://kurier.readthedocs.io/en/latest/#get) docs show what information is available in this context.

Reply:
@joelalejandro I got stuck with the test cases for this example. I want to implement some real use-cases in test-app, for example ‘article’ has many ‘comments’ or something similar with ‘book’ resource, but I can’t make it work. Can you help me with this?

Reply:
@isBatak Sorry for the delay in my response! It's been a busy week. Can you tell me in greater detail what seems to be the problem? I'd be happy to help.

Reply:
@joelalejandro I'm trying to get response like this compound document https://jsonapi.org/format/1.1/#document-compound-documents but I can't find any example in the docs how to make `article` hasMany `comments` and get this part in the response:
```
"relationships": {
      "comments": {
        "data": [
          { "type": "comments", "id": "5" },
          { "type": "comments", "id": "12" }
        ]
      }
    }
 ```
 I would like to have this example in the `test-app`.

Reply:
I think the request should be `/article/1?include=comments` - @spersico can you help me out here?

Reply:
Idk why but comments are currently exposed on User resource... so this will not work. 
I guess I need to redesign the DB a bit ...